test/controllers/bulk_controller_test.exs

```diff
@@ -0,0 +1,197 @@
+defmodule CoursePlanner.BulkControllerTest do
+  use CoursePlannerWeb.ConnCase
+
+  alias CoursePlanner.{Repo, Accounts.User}
+
+  import CoursePlanner.Factory
+
+  @file_path "/tmp/csv_test.txt"
+
+  setup(%{user_role: role}) do
+    user = insert(role)
+
+    conn =
+      Phoenix.ConnTest.build_conn()
+      |> assign(:current_user, user)
+    {:ok, conn: conn}
+  end
+
+  defp create_input_params(target, title, csv_data) do
+    File.touch!(@file_path)
```

Comment:
I think it would be better to create the file with `Plug.Upload.random_file!/1`
This creates the file in a proper temp place and removes it after the process dies

something like
```elixir
upload = Plug.Upload.random_file!("csv")
File.write!(upload.path, csv_data)
 %{"input" => %{"csv_file" => upload, "target" => target, "title" => title}}
```

Reply:
Done, but I had to modify it a bit cause just return a path.

Reply:
cool! I confused it, I thought it would return a `%Plug.Upload{}` struct